ibis/pyspark/compiler.py

```diff
@@ -10,7 +10,6 @@
 import ibis.common.exceptions as com
 import ibis.expr.datatypes as dtypes
 import ibis.expr.operations as ops
-import ibis.expr.types as ir
```

Comment:
Removing this import is what it's causing the errors. Can you restore it please?

Reply:
i already check that but in my code i got 

![2020-10-14-161833_2880x1620_scrot](https://user-images.githubusercontent.com/1095986/96035152-01241a00-0e39-11eb-92db-a572931b2655.png)

Reply:
i already execute the commands,  here is the log, there is nothing to commit because the changes that are generate by black are rollback when i exec the isort 




(ibis-dev) ➜  ibis git:(master) conda env update
Collecting package metadata (repodata.json): done
Solving environment: done


==> WARNING: A newer version of conda exists. <==
  current version: 4.8.3
  latest version: 4.8.5

Please update conda by running

    $ conda update -n base -c defaults conda



Downloading and Extracting Packages
pygit2-1.3.0         | 209 KB    | ###################################################################################### | 100% 
conda-build-3.20.4   | 551 KB    | ###################################################################################### | 100% 
ipython-7.18.1       | 1.1 MB    | ###################################################################################### | 100% 
matplotlib-3.3.2     | 6 KB      | ###################################################################################### | 100% 
asv-0.4.2            | 384 KB    | ###################################################################################### | 100% 
sqlalchemy-1.3.20    | 1.8 MB    | ###################################################################################### | 100% 
graphviz-2.42.3      | 7.3 MB    | ###################################################################################### | 100% 
isort-5.6.4          | 76 KB     | ###################################################################################### | 100% 
pre-commit-2.7.1     | 204 KB    | ###################################################################################### | 100% 
Preparing transaction: done
Verifying transaction: done
Executing transaction: done
Ran pip subprocess with arguments:
['/home/jelitox/.conda/envs/ibis-dev/bin/python', '-m', 'pip', 'install', '-U', '-r', '/home/jelitox/repos/lab/ibis/condaenv.3nj8v8h7.requirements.txt']
Pip subprocess output:
Collecting git+https://github.com/datapythonista/pysuerga (from -r /home/jelitox/repos/lab/ibis/condaenv.3nj8v8h7.requirements.txt (line 1))
  Cloning https://github.com/datapythonista/pysuerga to /tmp/pip-req-build-_kkrvoxp
Requirement already satisfied, skipping upgrade: markdown in /home/jelitox/.conda/envs/ibis-dev/lib/python3.8/site-packages (from pysuerga==0.1.0->-r /home/jelitox/repos/lab/ibis/condaenv.3nj8v8h7.requirements.txt (line 1)) (3.2.2)
Requirement already satisfied, skipping upgrade: jinja2 in /home/jelitox/.conda/envs/ibis-dev/lib/python3.8/site-packages (from pysuerga==0.1.0->-r /home/jelitox/repos/lab/ibis/condaenv.3nj8v8h7.requirements.txt (line 1)) (2.11.2)
Requirement already satisfied, skipping upgrade: pyyaml in /home/jelitox/.conda/envs/ibis-dev/lib/python3.8/site-packages (from pysuerga==0.1.0->-r /home/jelitox/repos/lab/ibis/condaenv.3nj8v8h7.requirements.txt (line 1)) (5.3.1)
Requirement already satisfied, skipping upgrade: MarkupSafe>=0.23 in /home/jelitox/.conda/envs/ibis-dev/lib/python3.8/site-packages (from jinja2->pysuerga==0.1.0->-r /home/jelitox/repos/lab/ibis/condaenv.3nj8v8h7.requirements.txt (line 1)) (1.1.1)
Building wheels for collected packages: pysuerga
  Building wheel for pysuerga (setup.py): started
  Building wheel for pysuerga (setup.py): finished with status 'done'
  Created wheel for pysuerga: filename=pysuerga-0.1.0-py3-none-any.whl size=9297 sha256=4260144fcc5047a20e7d37a09a54da3c2deebadc57db4bdf3da1d0c67267c588
  Stored in directory: /tmp/pip-ephem-wheel-cache-n9uprbko/wheels/f7/eb/9c/e99c5c4bc5b44c852e4e0a50af5d6e69b65920908d4dc9b12b
Successfully built pysuerga
Installing collected packages: pysuerga
  Attempting uninstall: pysuerga
    Found existing installation: pysuerga 0.1.0
    Uninstalling pysuerga-0.1.0:
      Successfully uninstalled pysuerga-0.1.0
Successfully installed pysuerga-0.1.0

#
# To activate this environment, use
#
#     $ conda activate ibis-dev
#
# To deactivate an active environment, use
#
#     $ conda deactivate


(ibis-dev) ➜  ibis git:(master) conda list -e |grep black           
black=19.10b0=py_4
(ibis-dev) ➜  ibis git:(master) conda list -e | grep isort 
isort=5.6.4=py_0
(ibis-dev) ➜  ibis git:(master) black .
reformatted /home/jelitox/repos/lab/ibis/ibis/__init__.py
reformatted /home/jelitox/repos/lab/ibis/ibis/impala/api.py
reformatted /home/jelitox/repos/lab/ibis/ibis/backends/postgres/compiler.py
All done! ✨ 🍰 ✨
3 files reformatted, 270 files left unchanged.
(ibis-dev) ➜  ibis git:(master) ✗ isort .
WARNING: Unable to parse file . due to [Errno 21] Is a directory: '/home/jelitox/repos/lab/ibis'
(ibis-dev) ➜  ibis git:(master) ✗ isort   
--- /home/jelitox/repos/lab/ibis/ibis/__init__.py:before	2020-10-14 16:41:23.865412
+++ /home/jelitox/repos/lab/ibis/ibis/__init__.py:after	2020-10-14 16:41:39.769500
@@ -5,7 +5,6 @@
 import ibis.expr.api as api  # noqa: F401
 import ibis.expr.types as ir  # noqa: F401
 import ibis.util as util  # noqa: F401
-
 # pandas backend is mandatory
 from ibis.backends import pandas  # noqa: F401
 from ibis.common.exceptions import IbisError  # noqa: F401
Apply suggested changes to '/home/jelitox/repos/lab/ibis/ibis/__init__.py' [y/n/q]? y
Fixing /home/jelitox/repos/lab/ibis/ibis/__init__.py
--- /home/jelitox/repos/lab/ibis/ibis/impala/api.py:before	2020-10-14 16:41:26.408746
+++ /home/jelitox/repos/lab/ibis/ibis/impala/api.py:after	2020-10-14 16:42:09.139453
@@ -1,6 +1,5 @@
 import ibis.common.exceptions as com
 from ibis.config import options
-
 # these objects are exposed in the public API and are not used in the module
 from ibis.impala.client import (  # noqa: F401
     ImpalaClient,
Apply suggested changes to '/home/jelitox/repos/lab/ibis/ibis/impala/api.py' [y/n/q]? y
Fixing /home/jelitox/repos/lab/ibis/ibis/impala/api.py
--- /home/jelitox/repos/lab/ibis/ibis/backends/postgres/compiler.py:before	2020-10-14 16:41:27.595412
+++ /home/jelitox/repos/lab/ibis/ibis/backends/postgres/compiler.py:after	2020-10-14 16:42:13.306697
@@ -19,7 +19,6 @@
 import ibis.expr.operations as ops
 import ibis.expr.types as ir
 import ibis.sql.alchemy as alch
-
 # used for literal translate
 from ibis.sql.alchemy import (
     _get_sqla_table,
Apply suggested changes to '/home/jelitox/repos/lab/ibis/ibis/backends/postgres/compiler.py' [y/n/q]? y
Fixing /home/jelitox/repos/lab/ibis/ibis/backends/postgres/compiler.py
Skipped 3 files

Reply:
I don't understand why isort should revert black changes. Black should only change formatting, and isort the order of the imports. Can you run them in opposite order? First isort and then black?

If you checked you've got the same versions as in the CI, can you change manually or do whatever until both `black` and `isort --checkonly` don't report any problem locally, and then commit please? We can't move forward until those errors are fixed.

Thanks!